source/extensions/stat_sinks/hystrix/hystrix.cc

```diff
@@ -0,0 +1,375 @@
+       ++stats_map_itr) {
+    out_str << stats_map_itr->first << " | ";
+    RollingWindow rolling_window = stats_map_itr->second;
+    for (auto specific_stat_vec_itr = rolling_window.begin();
+         specific_stat_vec_itr != rolling_window.end(); ++specific_stat_vec_itr) {
+      out_str << *specific_stat_vec_itr << " | ";
+    }
+    out_str << std::endl;
+  }
+  return out_str.str();
+}
+
+namespace Hystrix {
+HystrixSink::HystrixSink(Server::Instance& server)
+    : stats_(new HystrixStatCache()), server_(&server) {
+  Server::Admin& admin = server_->admin();
+  ENVOY_LOG(debug,
+            "adding hystrix_event_stream endpoint to enable connection to hystrix dashboard");
+  admin.addHandler("/hystrix_event_stream", "send hystrix event stream",
+                   MAKE_ADMIN_HANDLER(handlerHystrixEventStream), false, false);
+};
+
+Http::Code HystrixSink::handlerHystrixEventStream(absl::string_view,
+                                                  Http::HeaderMap& response_headers,
+                                                  Buffer::Instance&,
+                                                  Server::AdminFilter& admin_filter) {
+
+  response_headers.insertContentType().value().setReference(
+      Http::Headers::get().ContentTypeValues.TextEventStream);
+  response_headers.insertCacheControl().value().setReference(
+      Http::Headers::get().CacheControlValues.NoCache);
+  response_headers.insertConnection().value().setReference(
+      Http::Headers::get().ConnectionValues.Close);
+  response_headers.insertAccessControlAllowHeaders().value().setReference(
+      Http::Headers::get().AccessControlAllowHeadersValue.AccessControlAllowHeadersHystrix);
+  response_headers.insertAccessControlAllowOrigin().value().setReference(
+      Http::Headers::get().AccessControlAllowOriginValue.All);
+  response_headers.insertNoChunks().value().setReference("0");
+
+  Http::StreamDecoderFilterCallbacks* stream_decoder_filter_callbacks =
+      admin_filter.getDecoderFilterCallbacks();
+
+  registerConnection(stream_decoder_filter_callbacks);
+
+  // Separated out just so it's easier to understand
+  auto on_destroy_callback = [this, stream_decoder_filter_callbacks]() {
+    // Unregister the callbacks from the sink so data is no longer encoded through them.
+    unregisterConnection(stream_decoder_filter_callbacks);
+  };
+
+  // Add the callback to the admin_filter list of callbacks
+  admin_filter.addOnDestroyCallback(std::move(on_destroy_callback));
+
+  ENVOY_LOG(debug, "start sending data to hystrix dashboard on port {}",
+            stream_decoder_filter_callbacks->connection()->localAddress()->asString());
+  return Http::Code::OK;
+}
+
+void HystrixSink::beginFlush() { current_stat_values_.clear(); }
+
+void HystrixSink::flushCounter(const Stats::Counter& counter, uint64_t) {
+  if (callbacks_list_.empty()) {
+    return;
+  }
+  if (counter.name().find("upstream_rq_") != std::string::npos) {
```

Comment:
The existing struct do not cover all the stats we need. missing - `upstream_rq_2xx`, `upstream_rq_4xx`, `upstream_rq_5xx`. To get rid of the string compare, we'll need to read those directly in `end_flush()`. Or is it possible to create struct with only the stats we need?
By the way, we use the stats() method here: https://github.com/trabetti/envoy/blob/bb876ee141069371ad0624bc3efc08074a0d7602/source/extensions/stat_sinks/hystrix/hystrix.cc#L338

Reply:
That's a good point - my bad, those are actually created and set dynamically - https://github.com/envoyproxy/envoy/blob/b8f2268e6bd1dd2a7cb80bde64267dc7c8384768/source/common/http/codes.cc#L23.  For those, I would recommend looking them up dynamically  similar to how you currently look up the `.membership_total` stat.  

 The `stats()` method you are referring to is the method on the Server::Instance object, which returns a stats scope that allows looking up/creating stats by name.  I'm referring to the `stats()` method on the cluster object, which returns a struct of specific stats that are relevant to the cluster that can be accessed directly without a lookup.  The latter should be used for as many stats as possible as it is the cheapest lookup.  The former should be used for specific stats that are not accessible directly via a struct, which IIUC, should just be those 3 stats you just mentioned.  We can't really create our own struct for all of these stats because it would need to constantly be updated whenever Envoy adds or removes a cluster.

I would really only advise using the `flushCounter`, `flushGauge`, etc methods when you need to export all or most of Envoy's stats. It seems this sink is interested in a pretty limited subset of stats, which means individual lookups will likely be the cheapest way to do it.

I know the stats system can be pretty confusing because there are a lot of different ways to do the same thing :).  Does all of that make sense?

Reply:
Just for my own learnings....is there some reason an interested class can't do the name-lookup once on construction and then save the desired stat in their own member var?